src/main/java/com/sumologic/jenkins/jenkinssumologicplugin/sender/LogListener.java

```diff
@@ -0,0 +1,42 @@
+package com.sumologic.jenkins.jenkinssumologicplugin.sender;
+
+import com.sumologic.jenkins.jenkinssumologicplugin.PluginDescriptorImpl;
+import com.sumologic.jenkins.jenkinssumologicplugin.integration.SearchAction;
+import hudson.Extension;
+import hudson.console.ConsoleLogFilter;
+import hudson.model.AbstractBuild;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * This class intercepts console output stream for every jenkins build and decorates it with additional functionality.
+ * Depending on configuration it can add two functionalities:
+ * 1) Add consistent timestamps to all output lines.
+ * 2) Forward all output logs to Sumo Logic.
```

Comment:
It is not the case that Jenkins timestamps just have hours: they have minutes and seconds, e.g. https://jenkins.kumoroku.com/job/Master-Update-Jenkins/844/console.  No date as printed there, though I suspect it's in the sidecar data on disk.  (The timestamps are stored different from the raw log data.)

At any rate, as long as it only goes to Sumo, that's fine with me.  Personally, I wouldn't have them even optionally go elsewhere - this plugin should be focused on Sumo, not on providing other bits of functionality.